npm/src/upload/upload-form.ts

```diff
@@ -63,83 +135,117 @@ export class UploadForm {
-          ".drag-and-drop__failure"
-        )
-        warningMessage?.classList.add("hide")
-        const successMessage: HTMLElement | null = document.querySelector(
-          ".drag-and-drop__success"
-        )
-        successMessage?.classList.remove("hide")
-        this.dropzone.classList.remove("drag-and-drop__dropzone--dragover")
-      }
+  addFolderListener() {
+    this.dropzone.addEventListener("drop", this.handleDropppedItems)
+
+    this.folderRetriever.addEventListener("change", () => {
+      const form: HTMLFormElement | null = this.formElement
+      const files = this.retrieveFiles(form)
+      this.selectedFiles = files
+      const parentFolder = this.getParentFolderName(this.selectedFiles)
+      this.updateFolderSelectionStatus(parentFolder, files.length)
     })
   }
 
   addSubmitListener(
-    uploadFiles: (files: TdrFile[], uploadFilesInfo: FileUploadInfo) => void
+    uploadFiles: (
+      files: IFileWithPath[],
+      uploadFilesInfo: FileUploadInfo
+    ) => void
   ) {
     this.formElement.addEventListener("submit", (ev) => {
       ev.preventDefault()
-      const target: HTMLInputTarget | null = ev.currentTarget
-      const files = this.retrieveFiles(target)
-      const parentFolder = this.getParentFolderName(files)
+      const parentFolder = this.getParentFolderName(this.selectedFiles)
       const uploadFilesInfo: FileUploadInfo = {
         consignmentId: this.consignmentId(),
         parentFolder: parentFolder
       }
-      uploadFiles(files, uploadFilesInfo)
+      uploadFiles(this.selectedFiles, uploadFilesInfo)
     })
   }
 
-  private retrieveFiles(target: HTMLInputTarget | null): TdrFile[] {
-    const files: TdrFile[] = target!.files!.files!
+  private getParentFolderName(folder: IFileWithPath[]) {
+    const firstItem: FileWithRelativePath = folder[0]
+      .file as FileWithRelativePath
+    const relativePath: string = firstItem.webkitRelativePath
+    const splitPath: string[] = relativePath.split("/")
+    const parentFolder: string = splitPath[0]
+    return parentFolder
+  }
+
+  private checkIfFolderHasFiles(files: File[] | IFileWithPath[]): void {
     if (files === null || files.length === 0) {
-      this.dropzone.classList.remove("drag-and-drop__dropzone--dragover")
-      const successMessage: HTMLElement | null = document.querySelector(
-        ".drag-and-drop__success"
-      )
-      successMessage?.classList.add("hide")
+      this.rejectUserItemSelection()
+    }
+  }
+
+  private retrieveFiles(target: HTMLInputTarget | null): IFileWithPath[] {
+    const files: File[] = target!.files!.files!
+    this.checkIfFolderHasFiles(files)
+
+    return [...files].map((file) => ({
+      file,
+      path: (file as FileWithRelativePath).webkitRelativePath
+    }))
+  }
+
+  private rejectUserItemSelection() {
+    this.dropzone.classList.remove("drag-and-drop__dropzone--dragover") // Extract this into its own method to do with file selection
```

Comment:
This comment sounds like a to-do?

Reply:
Again, I'll let @techncl have final say but this is repeated a couple of times in the class with the same specific string for the class so it makes sense to have it as it's own function. I've moved it out but I can always put it back.

Reply:
Yes, you are both correct, I forgot to do it. I thought writing comments in my code would remind me to add the feature, but I guess I need a reminder to look at my reminders :smile: